conductor_api/src/interface.rs

```diff
@@ -138,74 +187,90 @@ impl ConductorApiBuilder {
-                match maybe_provenance {
-                    None => make_cap_request_for_call(
-                        context.clone(),
-                        token,
-                        &func_name,
-                        JsonString::from_json(&args_string.clone()),
-                    ),
-                    Some(json_provenance) => {
-                        let provenance: Provenance =
-                            serde_json::from_value(json_provenance.to_owned()).map_err(|e| {
-                                jsonrpc_core::Error::invalid_params(format!(
-                                    "invalid provenance: {}",
-                                    e
-                                ))
-                            })?;
-                        CapabilityRequest::new(token, provenance.source(), provenance.signature())
-                    }
-                }
-            };
+            let (zome_name, func_name, args_string) = Self::get_call_args(&params_map)?;
+            let cap_request = Self::get_cap_request(hc.context().unwrap(), &func_name, &params_map, &args_string)?;
 
             let response = hc
                 .call(&zome_name, cap_request, &func_name, &args_string)
                 .map_err(|e| jsonrpc_core::Error::invalid_params(e.to_string()))?;
+
             Ok(Value::String(response.to_string()))
         });
     }
 
+    /// Adds a call and measure function to the conductor. This is currently only available on linux
+    #[cfg(target_os = "linux")]
+    pub fn with_call_and_measure(mut self) -> Self {
+        let instances = self.instances.clone();
+        let instance_ids_map = self.instance_ids_map.clone();
+
+        self.io.add_method("call_and_measure", move |params| {
+            if cfg!(target_os = "linux") {
+                let params_map = Self::unwrap_params_map(params)?;
+                let public_id_str = Self::get_as_string("instance_id", &params_map)?;
+
+                let id = instance_ids_map
+                    .get(&PublicInstanceIdentifier::from(public_id_str))
+                    .ok_or(jsonrpc_core::Error::invalid_params(
+                        "instance identifier invalid",
+                    ))?;
+                let instance = instances
+                    .get(id)
+                    .ok_or(jsonrpc_core::Error::invalid_params("unknown instance"))?;
+
+                let hc_lock = instance.clone();
+                let hc_lock_inner = hc_lock.clone();
+                let mut hc = hc_lock_inner.write().unwrap();
+
+                let (zome_name, func_name, args_string) = Self::get_call_args(&params_map)?;
+                let cap_request = Self::get_cap_request(hc.context().unwrap(), &func_name, &params_map, &args_string)?;
+
+                // setup the self-meter in another thread to monitor this one
+                let mut meter = self_meter::Meter::new(Duration::from_millis(100)).unwrap();
```

Comment:
This is the core of it here. A monitor thread is started which scans every 100ms and computes a report when the call has finished which is averaged from the scans.